src/smcache/writer.rs

```diff
@@ -93,7 +93,7 @@ impl SmCacheWriter {
             DecodedMap::Hermes(smh) => smh.sources().zip(smh.source_contents()),
             DecodedMap::Index(_smi) => unreachable!(),
         }
-        .filter_map(|(name, source)| source.map(|source| (name, source)));
+        .map(|(name, source)| (name, source.unwrap_or_default()));
```

Comment:
I’m a bit confused. This `unwrap_or_default` means we write empty (but existing) strings into the smcache. Where exactly do these turn into `None` again?

Reply:
We don't write empty strings, the `insert_string` function just returns `u32::MAX` in that case, which is interpreted as `None` when reading.

Reply:
Ah yes, this makes sense. I wonder if this matters, probably not.